crates/matrix-sdk/src/room/mod.rs

```diff
@@ -403,6 +403,18 @@ impl Room {
         IdentityStatusChanges::create_stream(self.clone()).await
     }
 
+    /// Mark this user as "pinned" meaning that we have seen their cryptographic
+    /// identity before.
+    #[cfg(feature = "e2e-encryption")]
+    pub async fn pin_user_identity(&self, user_id: &UserId) -> Result<()> {
+        let x = self.client.encryption().get_user_identity(user_id).await?;
+
+        match x {
+            Some(x) => Ok(x.pin().await?),
+            None => Err(Error::UserNotFound),
+        }
+    }
```

Comment:
this doesn't seem specific to rooms... it seems a bit odd to expose it as part of `Room`?

Reply:
This was particularly requested by @stefanceriu - I guess it's for convenience because they have a room already available. I guess we could put it on `Client`. Any comment @stefanceriu ?

Reply:
Agreed this doesn't make sense to be on `Room`. Please let's not make the API mess solely for the purpose of convenience. `Client` does make sense.

Reply:
Yes, that was the reason. Also it makes it consistent with the other verification withdrawal methods. Otherwise you could even put it in the Client or even Encryption. Up to you really ¯\_(ツ)_/¯

Reply:
Changed to `Encryption::get_user_identity` then `UserIdentity::pin` in the force-pushed update I just did.